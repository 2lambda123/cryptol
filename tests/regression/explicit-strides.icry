[ 0:Integer .. 10 by 2]
[ 0 .. <10:Integer by 2]

[ 0 .. 9 by 2] : [_]Integer
[ 0 .. <9 by 2:Integer]

[ 10:Integer .. 0 down by 2 ]
[ 10 .. >0 down by 2 ]

[ 2^^16-1 .. >50000 down by 100 ] : [_][16]

[ 200 .. 100:Integer down by 9 ]
