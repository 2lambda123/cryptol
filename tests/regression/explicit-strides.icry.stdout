Loading module Cryptol
[0, 2, 4, 6, 8, 10]
[0, 2, 4, 6, 8]
[0, 2, 4, 6, 8]
[0, 2, 4, 6, 8]
[10, 8, 6, 4, 2, 0]
Showing a specific instance of polymorphic result:
  * Using 'Integer' for type argument 'a' of 'Cryptol::fromToDownByGreaterThan'
[10, 8, 6, 4, 2]
[0xffff, 0xff9b, 0xff37, 0xfed3, 0xfe6f, 0xfe0b, 0xfda7, 0xfd43,
 0xfcdf, 0xfc7b, 0xfc17, 0xfbb3, 0xfb4f, 0xfaeb, 0xfa87, 0xfa23,
 0xf9bf, 0xf95b, 0xf8f7, 0xf893, 0xf82f, 0xf7cb, 0xf767, 0xf703,
 0xf69f, 0xf63b, 0xf5d7, 0xf573, 0xf50f, 0xf4ab, 0xf447, 0xf3e3,
 0xf37f, 0xf31b, 0xf2b7, 0xf253, 0xf1ef, 0xf18b, 0xf127, 0xf0c3,
 0xf05f, 0xeffb, 0xef97, 0xef33, 0xeecf, 0xee6b, 0xee07, 0xeda3,
 0xed3f, 0xecdb, 0xec77, 0xec13, 0xebaf, 0xeb4b, 0xeae7, 0xea83,
 0xea1f, 0xe9bb, 0xe957, 0xe8f3, 0xe88f, 0xe82b, 0xe7c7, 0xe763,
 0xe6ff, 0xe69b, 0xe637, 0xe5d3, 0xe56f, 0xe50b, 0xe4a7, 0xe443,
 0xe3df, 0xe37b, 0xe317, 0xe2b3, 0xe24f, 0xe1eb, 0xe187, 0xe123,
 0xe0bf, 0xe05b, 0xdff7, 0xdf93, 0xdf2f, 0xdecb, 0xde67, 0xde03,
 0xdd9f, 0xdd3b, 0xdcd7, 0xdc73, 0xdc0f, 0xdbab, 0xdb47, 0xdae3,
 0xda7f, 0xda1b, 0xd9b7, 0xd953, 0xd8ef, 0xd88b, 0xd827, 0xd7c3,
 0xd75f, 0xd6fb, 0xd697, 0xd633, 0xd5cf, 0xd56b, 0xd507, 0xd4a3,
 0xd43f, 0xd3db, 0xd377, 0xd313, 0xd2af, 0xd24b, 0xd1e7, 0xd183,
 0xd11f, 0xd0bb, 0xd057, 0xcff3, 0xcf8f, 0xcf2b, 0xcec7, 0xce63,
 0xcdff, 0xcd9b, 0xcd37, 0xccd3, 0xcc6f, 0xcc0b, 0xcba7, 0xcb43,
 0xcadf, 0xca7b, 0xca17, 0xc9b3, 0xc94f, 0xc8eb, 0xc887, 0xc823,
 0xc7bf, 0xc75b, 0xc6f7, 0xc693, 0xc62f, 0xc5cb, 0xc567, 0xc503,
 0xc49f, 0xc43b, 0xc3d7, 0xc373]
[200, 191, 182, 173, 164, 155, 146, 137, 128, 119, 110, 101]
